discovery_engine_core/core/src/engine.rs

```diff
@@ -763,6 +763,10 @@ impl Engine {
             &self.state.user_interests,
         );
         if let UserReaction::Positive = reaction {
+            #[cfg(feature = "storage")]
+            let history = Some(self.storage.fetch_history().await?);
+            #[cfg(feature = "storage")]
+            let history = &history;
```

Comment:
:art: this could be condensed into one line
```suggestion
            #[cfg(feature = "storage")]
            let history = &self.storage.fetch_history().await?.into();
```